server/tests/test_browser_admin.py

```
@@ -681,6 +683,142 @@ def test_modqueue_does_not_storm_the_list_endpoint(admin_page):
+    panel.wait_for(state="visible", timeout=5000)
+    assert "Factory reset" in panel.inner_text()
+    assert "WIPES RUNTIME STATE" in panel.inner_text()
+
+    admin_page.locator(".admin-hud-modal__btn--cancel").click()
+    admin_page.wait_for_timeout(600)
+    assert calls == [], f"取消 factory reset 之後仍然送出了請求：{calls}"
+    assert dialogs == [], f"仍然跳出了原生對話框：{dialogs}"
+
+
+def test_revoke_fire_token_shows_hud_panel_and_cancel_sends_nothing(admin_page, live_url):
+    """撤銷 Fire Token：撤銷鍵要先有 token 才會啟用，取消必須什麼都不做。"""
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)